packages/@lwc/engine/src/framework/wc.ts

```diff
@@ -25,17 +27,21 @@ import { HTMLElementConstructor } from './base-bridge-element';
  * const elm = document.createElement('x-foo');
  * ```
  */
-export function buildCustomElementConstructor(
-    Ctor: ComponentConstructor,
-    options?: {
-        mode?: 'open' | 'closed';
-    }
+export function deprecatedBuildCustomElementConstructor(
+    Ctor: ComponentConstructor
 ): HTMLElementConstructor {
+    /* eslint-disable-next-line no-console */
+    console.error(
```

Comment:
Add this into a production check block to avoid leaking the `console.log` in production.

Reply:
I don't think we care much here... I'm fine with doing so, but in platform that method is not available... only in OSS.

Reply:
Even if it is not available, it is still shipped because there is no tree-shaking on the platform.